enaml/core/compiler_common.py

```diff
@@ -259,6 +259,11 @@ def load_name(cg, name, local_names):
     """
     if name in local_names:
         cg.load_fast(name)
+    elif '.' in name:
+        parts = name.split('.')
+        load_name(cg, parts.pop(0), local_names)
+        while parts:
+            cg.load_attr(parts.pop(0))
```

Comment:
This is the wrong place for this logic. The `name` argument is assumed to not have a dot. We should really make a new ast node to describe the dotted  type name, then do a more granular emit in the compiler.

Hiding the logic away in this function, while easy, is a bit too much magic for my taste.